docs/local_dev.md

```diff
@@ -179,3 +179,83 @@ kubectl delete cluster $CLUSTER_NAME
 docker rm -f $HOST_NAME
 kind delete cluster
 ```
+
+# Host Agent Installer
+The installer is responsible for detecting the BYOH OS, downloading a BYOH bundle and installing/uninstalling it.
+
+## Creating a BYOH Bundle
+### Upstream Ingredients
+Optional. This step describes downloading upstream k8s host components for Debian.
+```shell
+#Build docker image
+(cd agent/installer/bundle_builder/ingredients/deb/ && docker build -t byoh-ingredients-deb .)
+
+#Create a directory for the ingredients and download to it
+(mkdir -p byoh-ingredients-download && docker run --rm -v `pwd`/byoh-ingredients-download:/ingredients byoh-ingredients-deb)
+```
+### Downstream Ingredients
+This step describes providing custom k8s host components. They can be copied to to byoh-ingredients-download. Files must match the following globs:
+```shell
+*containerd*.tar
+*kubeadm*.deb
+*kubelet*.deb
+*kubectl*.deb
+*cri-tools*.deb
+*kubernetes-cni*.deb
+```
+
+## Building a BYOH Bundle
+```shell
+#Build docker image
+(cd agent/installer/bundle_builder/ && docker build -t byoh-build-push-bundle .)
+```
+
+```shell
+#Build a BYOH bundle and publish it to an OCI-compliant repo
```

Comment:
Also worth adding how to push to a local registry? wdyt?

Reply:
Pushing to local registry should be like pushing to non-local.

If there is something very special in pushing, like we need, you can use the BUILD_ONLY=0 which is already explained.